core/src/androidMain/kotlin/com/beepiz/bluetooth/gattcoroutines/GattConnectionImpl.kt

```diff
@@ -276,7 +276,7 @@ internal class GattConnectionImpl(
         }
 
         override fun onCharacteristicChanged(gatt: BG, characteristic: BGC) {
-            launch { characteristicChangedChannel.send(characteristic) }
+            runBlocking { characteristicChangedChannel.send(characteristic) }
```

Comment:
How is blocking a thread helping here?

Did you try with #53 instead?

Reply:
That's an awesome question, but I don't have any answer. This just works, rather we like it or not.

Reply:
> Did you try with #53 instead?

Yes, in fact I thought that had solved the issue, but turns out nothing but this solved this crazy behavior.